### DISCUSSION TITLE
Table of contents

### DISCUSSION THREAD

**Discussion by rmccar** [ONSdigital/design-system#1382] [Feedback]
Use this place to discuss the [table of contents component](https://ons-design-system.netlify.com/components/toc/) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by trevorsaint** [comment]
Ensure spacing between each item is consistent.

This bug is due to the extra padding caused when using `display: inline-block` on the link. Adding an attribute `vertical-align: top` will fix this.

Currently this:
![image](https://user-images.githubusercontent.com/4989027/87784326-f6ec8280-c82d-11ea-9535-e8beaf071c21.png)

Should be this:
![image](https://user-images.githubusercontent.com/4989027/87784355-079cf880-c82e-11ea-8f05-ead09fe76da5.png)

**Reply by jrbarnes9** [reply]
#844 resolved

**Comment by jrbarnes9** [comment]
Propose to create a variant of ToC component which can be used as a "sticky" side nav.

To work in the same way as: https://www.gov.uk/service-manual/design/introduction-designing-government-services#the-characteristics-of-a-good-government-service

<img width="536" alt="Screenshot 2021-08-31 at 12 55 38" src="https://user-images.githubusercontent.com/43346934/131501109-82797071-1ef2-47d5-98a4-7b08cf686833.png">

**Reply by jrbarnes9** [reply]
Issue #1654 raised

**Comment by mr-nick17** [comment]
The example contains broken html; there's a closing `</aside>` tag but no opening. I've implemented the sticky TOC on the ONS website and have added the containing css class `ons-toc-container` to the `<aside>` but I am essentially overriding all the properties within: 
```scss 
.ons-toc-container {
  border-bottom: none;
  margin-bottom: 0;
  padding-bottom: 0;
}

**Reply by jrbarnes9** [reply]
Raised #1790

**Reply by jrbarnes9** [reply]
@mr-nick17 Are you setting `skipLink`? This is a required object, and the opening `<aside>` won't render without it.

**Reply by jrbarnes9** [reply]
Actually, I don't think `skipLink` should be used for sticky navs as they're in-page links, so the keyboard user would only need to pass over them once.  I'll remove this from the logic

**Reply by mr-nick17** [reply]
I am lifting the generated html directly into a Go template, so not using the nunjucks logic. I just noticed that the html was broken in the example. I was wondering about the skip links in the sticky nav too, I'll remove it from our website implementation

**Comment by Al-Davies** [comment]
**UR findings on table of contents**
tested as part of Census website testing Nov-Dec 2020

- TOC's have been used successfully as part of the 2021 Census website
- users understanding that these will jump them down to the relevant sections of the page
- where floating TOC's are used (in statistical bulletins), users don't always notice these
- users may use them to jump down to sections, but generally scroll back up the page to get to the top (not often seeing any back to top links)

**Comment by Joe-Dixon-ID** [comment]
The sticky table of contents (In-page table of contents variant) test really well in UR sessions but on the website we have no control over the content that is used on it (i.e it could make the height greater that the viewport height which makes degrades the UX of the component). 

I want to propose a version that is non sticky by default but uses javascript to insert the class that makes the table sticky. 

This way it scrolls with the page (non-sticky) if its greater than viewport height (or even if users has JS off) and if not it uses the sticky version. 

So if you remove the class ons-grid__col--sticky@m in the HTML and have JS like this for example:

document.addEventListener("DOMContentLoaded", function () {
  function updateStickyToc() {
    const tocContainer = document.querySelector(".ons-grid__col");
    const toc = document.querySelector(".ons-toc-container");

    if (toc && tocContainer) {
      const tocHeight = toc.offsetHeight;
      const viewportHeight = window.innerHeight;

      if (tocHeight <= viewportHeight) {
        tocContainer.classList.add("ons-grid__col--sticky@m");
      } else {
        tocContainer.classList.remove("ons-grid__col--sticky@m");
      }
    }
  }

  // Run on page load
  updateStickyToc();

  // Run on window resize
  window.addEventListener("resize", updateStickyToc);
});

**Reply by MagdalenaLarge** [reply]
Hello @Joe-Dixon-ID your request has been reviewed and added to our backlog. You can watch the progress of this ticket: https://jira.ons.gov.uk/browse/ONSDESYS-319. Thank you.

**Comment by sanjeevz3009** [comment]
### Table of Contents Styling Issue

The issue surfaces when you only have one item in the `itemsList`, where, when the page is rendered, the ToC single item seems to de-dent itself.

Current behaviour:
<img width="346" height="133" alt="Screenshot 2026-02-03 at 11 33 49" src="https://github.com/user-attachments/assets/51d87696-8d55-4db0-b611-fcc646bb0c27" />


Expected behaviour:
<img width="322" height="156" alt="Screenshot 2026-02-03 at 11 33 16" src="https://github.com/user-attachments/assets/0ac8c9b2-597b-4ffe-8109-0f1f7e9051d1" />

Thanks!

**Reply by MagdalenaLarge** [reply]
Hi @sanjeevz3009 , thank you for highlighting it out to our team. I will keep you posted about the progress.

**Reply by sanjeevz3009** [reply]
Thanks @MagdalenaLarge

**Reply by MagdalenaLarge** [reply]
Hello @sanjeevz3009,
thank you for your patience. I can confirm that issue has been resolved and it will be included in the next Design System release. Thank you.

**Reply by sanjeevz3009** [reply]
Thanks @MagdalenaLarge
